docs/src/idiomatic/007if-then-else.md

```diff
@@ -0,0 +1,94 @@
+# Idiom 7: Use Boolean operators in actions, not `IF-THEN-ELSE`
+
+## Description
+
+TLA+ provides an `IF-THEN-ELSE` operator, and it can be pretty tempting to use it for flow control as it's done in procedural programming. However, TLA+ is about transitions over a state machine, and a transition-defining action declared with `IF-THEN-ELSE` can be more complex than 2 actions declared without it. Considering that any expression in the form `IF b THEN x ELSE y` with `x` and `y` being booleans (and most TLA+ expressions are booleans) can be rewritten as `(b /\ x) \/ (!b /\ y)`, there's a pattern we can apply to get rid of some potentially troubled `IF-THEN-ELSE` definitions.
+
+The `IF-THEN-ELSE` operator can be used either to define a value or to branch some action as a sort of flow control. Defining values with `IF-THEN-ELSE` is common practice and is similar to the use of `IF` expressions on declarative programming languages. However, flow control in TLA+ can be done naturally by behavior definition through actions, making the use of `IF-THEN-ELSE` for flow control unnecessary. This idiom aims to clarify different usages of `IF-THEN-ELSE` expressions keeping in mind the TLA+ essence of declaring actions to define sets of transitions. 
```

Comment:
```suggestion
The `IF-THEN-ELSE` operator can be used either to define a value, or to branch some action as a sort of flow control. Defining values with `IF-THEN-ELSE` is common practice and is similar to the use of `IF` expressions in declarative programming languages. However, flow control in TLA+ can be done naturally by behavior definition through actions, making the use of `IF-THEN-ELSE` for flow control unnecessary. This idiom aims to clarify different usages of `IF-THEN-ELSE` expressions, keeping in mind the TLA+ essence of declaring actions to define transitions. 
```

Reply:
I meant it in the strict sense, buy your example is actually a good one to be included in the "When to use" section